### ISSUE TITLE
Pass channel to CoAPServer handle

### ISSUE THREAD

**Issue by vervaekejonathan** [Covertness/coap-rs#47]
I've been trying to add a channel to the coapserver message handler, but failing
This is as far as I get:

```
use coap::{CoAPRequest, CoAPResponse, CoAPServer};

use std::sync::mpsc;
use std::sync::mpsc::{Receiver, Sender};

fn request_handler(request: CoAPRequest, channel_tx : Sender<u8>) -> Option<CoAPResponse> {
    None
}

fn main() {
    let addr = "[::]:5683";

    let mut server = CoAPServer::new(addr).unwrap();
    
    let (channel_tx, channel_rx): (Sender<u8>, Receiver<u8>) = mpsc::channel();   
    server.handle( move |request| {
        request_handler(request, channel_tx.clone())
    }).unwrap();
}
```

Errors when building: 

```
error[E0525]: expected a closure that implements the `Fn` trait, but this closure only implements `FnOnce`
  --> src/main.rs:18:20
   |
18 |     server.handle( move |request| {
   |            ------  ^^^^^^^^^^^^^^ this closure implements `FnOnce`, not `Fn`
   |            |
   |            the requirement to implement `Fn` derives from here
19 |         request_handler(request, channel_tx)
   |                                  ---------- closure is `FnOnce` because it moves the variable `channel_tx` out of its environment

error[E0277]: `std::sync::mpsc::Sender<u8>` cannot be shared between threads safely
  --> src/main.rs:18:12
   |
18 |     server.handle( move |request| {
   |            ^^^^^^ `std::sync::mpsc::Sender<u8>` cannot be shared between threads safely
   |
   = help: within `[closure@src/main.rs:18:20: 20:6 channel_tx:std::sync::mpsc::Sender<u8>]`, the trait `std::marker::Sync` is not implemented for `std::sync::mpsc::Sender<u8>`
   = note: required because it appears within the type `[closure@src/main.rs:18:20: 20:6 channel_tx:std::sync::mpsc::Sender<u8>]`
   = note: required because of the requirements on the impl of `coap::server::CoAPHandler` for `[closure@src/main.rs:18:20: 20:6 channel_tx:std::sync::mpsc::Sender<u8>]`

error[E0277]: the trait bound `std::sync::mpsc::Sender<u8>: std::marker::Copy` is not satisfied in `[closure@src/main.rs:18:20: 20:6 channel_tx:std::sync::mpsc::Sender<u8>]`
  --> src/main.rs:18:12
   |
18 |     server.handle( move |request| {
   |            ^^^^^^ within `[closure@src/main.rs:18:20: 20:6 channel_tx:std::sync::mpsc::Sender<u8>]`, the trait `std::marker::Copy` is not implemented for `std::sync::mpsc::Sender<u8>`
   |
   = note: required because it appears within the type `[closure@src/main.rs:18:20: 20:6 channel_tx:std::sync::mpsc::Sender<u8>]`
   = note: required because of the requirements on the impl of `coap::server::CoAPHandler` for `[closure@src/main.rs:18:20: 20:6 channel_tx:std::sync::mpsc::Sender<u8>]`

error: aborting due to 3 previous errors
```

**Comment by vervaekejonathan** [comment]
that would be great! thanks!

**Comment by vervaekejonathan** [comment]
Any news on this? Thanks!

**Comment by Covertness** [comment]
finished most of them. i think a new release would out in this weekend.

**Comment by Covertness** [comment]
fixed in 0.8.0. please note the new release is based on nightly rust. because the async/await is still not available in stable rust. feel free to contact me if you have new question.
